### PR TITLE
chore: Write release notes for 0.28.0

### PR DESCRIPTION
**Pull request by eseidel**
Update RELEASE_NOTES.md for 0.28.0

### CODE REVIEW

**Review by eseidel — COMMENTED** [review]
This isn't ready for landing, since the new flutter engine isn't merged in, but I thought I'd get a head start on the release notes at least.

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -2,6 +2,31 @@
...
+   to help diagnose any later issues during patching.
+
+Shorebird's Flutter 3.19.4 contains a re-write of our Dart build pipeline on iOS
+which eliminates any slowdowns caused by patching Dart code in most cases. We
+built a new mechanism to make patched code build much more similarly to
+unpatched code, thus allowing more code sharing between the "patch" and
+"release" builds resulting in much faster execution at runtime.
+
+Most patches on iOS should now run at speeds indistinguishable from releases.
+If the tool-chain detects a case where the patch will run slower, it will
```
Comment (bryanoltman) [inline RELEASE_NOTES.md]:
```suggestion
If the toolchain detects a case where the patch will run slower, it will
```

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -2,6 +2,31 @@
...
+provide a warning. If you see any of these warnings, please let us know!
+
+Note: `shorebird patch ios` now takes a little longer as it has to do
+more work when building a patch.
+
+This is probably our last iOS beta before we declare Shorebird Code Push 1.0.
+We still have one more slowdown for un-patched builds we need to fix, but we
```
Comment (bryanoltman) [inline RELEASE_NOTES.md]:
```suggestion
We still have one more slowdown for unpatched builds we need to fix, but we
```

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -2,6 +2,31 @@
 
 This section contains past updates we've sent to customers via Discord.
 
+## 0.28.0 (March 27, 2024)
```
Comment (bryanoltman) [inline RELEASE_NOTES.md]:
```suggestion
## 0.28.0 (March 28, 2024)
```

**Review by bryanoltman — APPROVED** [review]

**Pull request merged**